common/gyro-dashboard.cpp

```diff
@@ -0,0 +1,54 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#include "motion-dashboard.h"
+
+using namespace rs2;
+using namespace rsutils::string;
+
+gyro_dashboard::gyro_dashboard( std::string name )
+    : motion_dashboard( name )
+{
+}
+
+void gyro_dashboard::process_frame( rs2::frame f )
+{
+    write_shared_data(
+        [&]()
+        {
+            if( f && f.is< rs2::motion_frame >()
+                && ( f.as< rs2::motion_frame >() ).get_profile().stream_type() == RS2_STREAM_GYRO )
+            {
+                double ts = glfwGetTime();
+                auto it = frame_to_time.find( static_cast<float>( f.get_profile().unique_id() ) );
+
+                if( ts - last_time > frame_rate && it != frame_to_time.end() )
+                {
+                    rs2::motion_frame frame = f.as< rs2::motion_frame >();
+
+                    x_value = frame.get_motion_data().x;
```

Comment:
I think that collecting all the data all the time is unnecessary. We should collect only the chosen displayed info. If the user changes the selection or the dashboard frame rate we can clear the history and start a new plot.

Reply:
I will change it. I sought that data collection of all parameters is not expensive for us.